cmd/node-termination-handler.go

```diff
@@ -291,7 +291,7 @@ func drainOrCordonIfNecessary(interruptionEventStore *interruptioneventstore.Sto
 		metrics.NodeActionsInc("pre-drain", nodeName, err)
 	}
 
-	if nthConfig.CordonOnly || drainEvent.IsRebalanceRecommendation() {
+	if nthConfig.CordonOnly || drainEvent.IsRebalanceRecommendation() && !nthConfig.DrainOnRebalance {
```

Comment:
Were you able to test this with `QueueProcessor` mode? The events to listen for are driven by EventBridge rules, so the basic setup for `QueueProcessor` would be:

```
helm upgrade --install aws-node-termination-handler \
  --namespace kube-system \
  --set enableSqsTerminationDraining=true \
  --set queueURL=https://sqs.us-east-1.amazonaws.com/0123456789/my-term-queue \
  eks/aws-node-termination-handler
```

However, this implementation looks like users would need to register EventBridge Rule for RebalanceRecommendation AND apply the config in the helm install command, which is kinda awkward relative to other events:

```
helm upgrade --install aws-node-termination-handler \
  --namespace kube-system \
  --set enableSqsTerminationDraining=true \
  --set drainOnRebalance=true \
  --set queueURL=https://sqs.us-east-1.amazonaws.com/0123456789/my-term-queue \
  eks/aws-node-termination-handler
```

Could you confirm this is the behavior? I'll be trying to think of a cleaner way in the meantime.

Reply:
Can you add parens: 

```
if nthConfig.CordonOnly || (drainEvent.IsRebalanceRecommendation() && !nthConfig.DrainOnRebalance) {
```
I realize it's the same when executed, but it's clearer with the parens.